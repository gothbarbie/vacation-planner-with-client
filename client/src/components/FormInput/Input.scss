
@import 'components/colors';

.input {
  display: flex;
  flex-direction: column;
  margin: 2rem 0;
  min-width: 220px;
}

.input__input {
  padding: .5rem 1rem;
  font-family: 'Avenir Next', sans-serif;
  font-size: 18px;
  color: $color-mine-shaft;
  border: 1px solid $color-silver;
  background: $color-white;
}

.input__input:focus {
  border: 1px solid $color-anakiwa;
}

.input__label {
  font-family: 'Avenir Next', sans-serif;
  font-size: 14px;
  color: $color-mine-shaft;
  margin-bottom: 1rem;
}

.input__label span {
  color: $color-burnt-sienna;
}

.input__error {
  background: $color-burnt-sienna;
  color: $color-white;
  font-family: 'Avenir Next', sans-serif;
  font-size: 14px;
  margin-top: .5rem;
  padding: .5rem;
  position: relative;
}

.input__error span {
  position: absolute;
  width: 0;
  height: 0;
  border-style: solid;
  border-width: 0 10px 10px 10px;
  border-color: transparent transparent $color-burnt-sienna transparent;
  top: -.45rem;
  left: 10px;
}
