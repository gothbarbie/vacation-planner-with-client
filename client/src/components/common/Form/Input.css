.input {
  display: flex;
  flex-direction: column;
  margin: 2rem 0;
}

.input__input {
  padding: .5rem 1rem;
}

.input__label {
  font-family: 'Avenir Next', sans-serif;
  font-size: 14px;
  color: #333;
  margin-bottom: 1rem;
}
