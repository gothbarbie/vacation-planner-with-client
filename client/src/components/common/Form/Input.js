// @flow
import React from 'react'

import './Input.css'

type props = {
  label: string,
  name: string,
  placeholder: string,
  error?: string
}

export default ({ error, name, placeholder, label }: props) => (
  <div className="input">
    <label className="input__label">{label}</label>
    <input className="input__input" name={name} placeholder={placeholder} type="text" />
    {error && <div className="input__error">{error}<span /></div>}
  </div>
)
