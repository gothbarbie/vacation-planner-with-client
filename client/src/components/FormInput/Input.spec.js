import React from 'react'
import { shallow } from 'enzyme'

import Input from './Input'

describe('components/Input', () => {
  const input = shallow(<Input />)

  it('renders', () => {
    expect(input).toMatchSnapshot()
  })
})
