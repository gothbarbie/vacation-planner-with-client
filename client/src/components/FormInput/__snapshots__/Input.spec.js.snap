// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input renders 1`] = `
<div
  className="input"
>
  <label
    className="input__label"
  />
  <input
    className="input__input"
    type="text"
  />
</div>
`;
